core/shared/src/main/scala/zio/Schedule.scala

```diff
@@ -1126,6 +1127,34 @@ object Schedule {
     Schedule(loop(None, 0L))
   }
 
+  /**
+   * Cron-like schedule that recurs every specified `minute` of each hour.
+   * Exact second of the minute can be provided as optional `second` param.
+   * Producing a count of repeats: 0, 1, 2.
+   */
+  def minuteOfHour(minute: Int, second: Int = 0): Schedule[Any, Any, Long] = {
+    assert(minute < 60 && minute >= 0, s"Invalid minute parameter. Must be in range 0 ... 59")
```

Comment:
If you are going to throw here, you need to wrap the whole function in `ZIO.effectSuspend`, so they will be translated into `ZIO.die`.

Reply:
that's kind of tricky. We can't wrap it with `ZIO.effectSuspend`, because the constructor is supposed to return `Schedule`. It doesn't seem that constructors are able to fail at all.

Reply:
Probably solution for this and also for #3850 would be warm-up run of `next` function inside Schedule#driver with `ref` change and `sleep` logic disabled. In this case placing `assert` inside `StepFunction` would be natural.

Reply:
Ah, you're right, we can't use effect suspend here. Maybe you should just clamp the values to `[0, 59]` or use wraparound? The alternative is to delay the failure until the schedule is used, which of course is possible but doesn't seem obviously better than clamping/wrapping.

Reply:
Could not think of anything better than use wraparound. We can clamp it in dotty latter using union and literal types.
The wrapping is still unsafe, so user will be responsible for catching errors.